File: Vanilla/sources/zdoc/media/lua/shared/Library/Item_Type.lua
---@class Item.Type : zombie.scripting.objects.Item.Type
---@field public Normal Item.Type
---@field public Weapon Item.Type
---@field public Food Item.Type
---@field public Literature Item.Type
---@field public Drainable Item.Type
---@field public Clothing Item.Type
---@field public Container Item.Type
---@field public WeaponPart Item.Type
---@field public Key Item.Type
---@field public KeyRing Item.Type
---@field public Moveable Item.Type
---@field public Radio Item.Type
---@field public AlarmClock Item.Type
---@field public AlarmClockClothing Item.Type
---@field public Map Item.Type
Item_Type = {}

---Returns an array containing the constants of this enum type, in
---
---the order they are declared.  This method may be used to iterate
---
---over the constants as follows:
---
---
---
---for (Item.Type c : Item.Type.values())
---
---    System.out.println(c);
---
---
---@public
---@return Item.Type[] @an array containing the constants of this enum type, in the order they are declared
function Item_Type:values() end

---Returns the enum constant of this type with the specified name.
---
---The string must match exactly an identifier used to declare an
---
---enum constant in this type.  (Extraneous whitespace characters are
---
---not permitted.)
---@public
---@param name String @the name of the enum constant to be returned.
---@return Item.Type @the enum constant with the specified name
function Item_Type:valueOf(name) end

File: Vanilla/sources/zomboid/zombie/inventory/ItemType.java
package zombie.inventory;

public enum ItemType {
   None(0),
   Weapon(1),
   Food(2),
   Literature(3),
   Drainable(4),
   Clothing(5),
   Key(6),
   KeyRing(7),
   Moveable(8),
   AlarmClock(9),
   AlarmClockClothing(10);

   private int index;

   private ItemType(int var3) {
      this.index = var3;
   }

   public int index() {
      return this.index;
   }

   public static ItemType fromIndex(int var0) {
      return ((ItemType[])ItemType.class.getEnumConstants())[var0];
   }

   // $FF: synthetic method
   private static ItemType[] $values() {
      return new ItemType[]{None, Weapon, Food, Literature, Drainable, Clothing, Key, KeyRing, Moveable, AlarmClock, AlarmClockClothing};
   }
}
